ib_edavki.py

```diff
@@ -166,13 +167,14 @@ def main():
         "isResident": taxpayer.find("isResident").text,
     }
 
-    """ Fetch companies.xml from GitHub if it doesn't exist and use the data for Doh-Div.xml """
+    """ Fetch companies.xml from GitHub if it doesn't exist or hasn't been updated for a month and use the data for Doh-Div.xml """
     companies = {}
-    if not os.path.isfile("companies.xml"):
-        urllib.request.urlretrieve(
+    if not os.path.isfile("companies.xml") or datetime.datetime.fromtimestamp(os.path.getctime("companies.xml")) < (datetime.datetime.now() - datetime.timedelta(days=30)):
```

Comment:
@jamsix @pronebird I don't know if I like this part too much. If a user has modified their local `companies.xml`, it would get overwritten. I think we should prompt the user for this or set up a way to use custom companies data.

Reply:
That's a valid point. 

I think it could be beneficial to separate user-defined companies from `companies.xml` and maybe introduce another file called `companies-overrides.xml` where users would add new companies. 

Both files can be merged together at the time of building a dictionary of companies. Then we don't have to worry about overwriting `companies.xml`.

Code-wise that would simply introduce a loop around here: https://github.com/jamsix/ib-edavki/blob/master/ib_edavki.py#L178

For example:

```python
for xml_name in ["companies.xml", "companies-overrides.xml"]:
  if os.path.isfile(xml_name):
        cmpns = xml.etree.ElementTree.parse(xml_name).getroot()
        for company in cmpns:
            c = {
                "symbol": company.find("symbol").text,
                "name": company.find("name").text,
                "taxNumber": company.find("taxNumber").text,
                "address": company.find("address").text,
                "country": company.find("country").text,
            }
            companies[c["symbol"]] = c
```

The only problem is that folks already modify `companies.xml` and I am not sure if all changes made it back to us. So maybe this can be solved by prompting the user to update and at the same time instruct them to use overrides file if they want to add or modify company data. We could also backup `companies.xml` before updating to `companies.xml.bak` file to preserve any user modifications just in case.